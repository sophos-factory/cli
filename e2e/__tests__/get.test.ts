/* eslint-disable @typescript-eslint/no-non-null-assertion */

import { format } from 'date-fns';
import isArray from 'lodash/isArray';
import { execute, executeAsIs } from '../helpers/execute';
import knownIds from '../helpers/knownIds';
import { withCmd } from '../helpers/options';
import yaml from 'js-yaml';

describe('refactrctl get', () => {
  jest.setTimeout(60 * 1000);

  test('throws on missing subcommand', async () => {
    await expect(
      execute(['get'], { token: process.env.FACTORY_STATIC_AUTH_TOKEN! })
    ).rejects.toMatchSnapshot();
  });

  test('formats as JSON when --format=json is passed', async () => {
    const formatJsonOrg = async () => {
      const json = JSON.stringify(
        await execute(
          [
            'get',
            'organization',
            knownIds.static.organization,
            '--format',
            'json'
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      );
      JSON.parse(json);
    };
    expect(formatJsonOrg).not.toThrow();

    const formatJsonProj = async () => {
      const json = JSON.stringify(
        await execute(
          ['get', 'project', knownIds.static.project, '--format', 'json'],
          {
            token: process.env.FACTORY_STATIC_AUTH_TOKEN!
          }
        )
      );
      JSON.parse(json);
    };
    expect(formatJsonProj).not.toThrow();

    const formatJsonCred = async () => {
      const json = JSON.stringify(
        await execute(
          [
            'get',
            'credential',
            knownIds.static.credential,
            '--project-id',
            knownIds.static.project,
            '--format',
            'json'
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      );
      JSON.parse(json);
    };
    expect(formatJsonCred).not.toThrow();

    const formatJsonJob = async () => {
      const json = JSON.stringify(
        await execute(
          [
            'get',
            'job',
            knownIds.static.job,
            '--project-id',
            knownIds.static.project,
            '--format',
            'json'
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      );
      JSON.parse(json);
    };
    expect(formatJsonJob).not.toThrow();

    const formatJsonPipe = async () => {
      const json = JSON.stringify(
        await execute(
          [
            'get',
            'pipeline',
            knownIds.static.pipeline,
            '--project-id',
            knownIds.static.project,
            '--format',
            'json'
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      );
      JSON.parse(json);
    };
    expect(formatJsonPipe).not.toThrow();
  });

  test('formats as YAML when --format=yaml is passed', async () => {
    const formatYamlOrg = async () => {
      const res = await execute(
        [
          'get',
          'organization',
          knownIds.static.organization,
          '--format',
          'yaml'
        ],
        { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
      );
      yaml.load(res);
    };
    expect(formatYamlOrg).not.toThrow();

    const formatYamlProj = async () => {
      const res = await execute(
        ['get', 'project', knownIds.static.project, '--format', 'yaml'],
        {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        }
      );
      yaml.load(res);
    };
    expect(formatYamlProj).not.toThrow();

    const formatYamlCred = async () => {
      const res = await execute(
        [
          'get',
          'credential',
          knownIds.static.credential,
          '--project-id',
          knownIds.static.project,
          '--format',
          'yaml'
        ],
        { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
      );
      yaml.load(res);
    };
    expect(formatYamlCred).not.toThrow();

    const formatYamlJob = async () => {
      const res = await execute(
        [
          'get',
          'job',
          knownIds.static.job,
          '--project-id',
          knownIds.static.project,
          '--format',
          'yaml'
        ],
        { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
      );
      yaml.load(res);
    };
    expect(formatYamlJob).not.toThrow();

    const formatYamlPipe = async () => {
      const res = await execute(
        [
          'get',
          'pipeline',
          knownIds.static.pipeline,
          '--project-id',
          knownIds.static.project,
          '--format',
          'yaml'
        ],
        { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
      );
      yaml.load(res);
    };
    expect(formatYamlPipe).not.toThrow();
  });

  describe('organization', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'organization'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });

    test('accepts id as argument', async () => {
      await expect(
        execute(['get', 'organization', knownIds.static.organization], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).resolves.toMatchSnapshot();
    });

    test('accepts id from stdin', async () => {
      await expect(
        executeAsIs(
          `echo ${knownIds.static.organization} | ${withCmd(
            'get organization'
          )}`,
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('project', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'project'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });

    test('accepts id as argument', async () => {
      await expect(
        execute(['get', 'project', knownIds.static.project], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).resolves.toMatchSnapshot();
    });

    test('accepts id from stdin', async () => {
      await expect(
        executeAsIs(
          `echo ${knownIds.static.project} | ${withCmd('get project')}`,
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('credential', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'credential'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(
          ['get', 'credential', '--project-id', knownIds.static.project],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).rejects.toMatchSnapshot();
    });

    test('accepts id as argument', async () => {
      await expect(
        execute(
          [
            'get',
            'credential',
            '--project-id',
            knownIds.static.project,
            knownIds.static.credential
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });

    test('accepts id from stdin', async () => {
      await expect(
        executeAsIs(
          `echo ${knownIds.static.credential} | ${withCmd(
            `get credential --project-id ${knownIds.static.project}`
          )}`,
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('job', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'job'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(['get', 'job', '--project-id', knownIds.static.project], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });

    test('accepts id as argument', async () => {
      await expect(
        execute(
          [
            'get',
            'job',
            '--project-id',
            knownIds.static.project,
            knownIds.static.job
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });

    test('accepts id from stdin', async () => {
      await expect(
        executeAsIs(
          `echo ${knownIds.static.job} | ${withCmd(
            `get job --project-id ${knownIds.static.project}`
          )}`,
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('pipeline-revision', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'pipeline-revision'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(
          [
            'get',
            'pipeline-revision',
            '--project-id',
            knownIds.static.project,
            '--pipeline-id',
            knownIds.static.pipeline
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).rejects.toMatchSnapshot();
    });

    test('accepts id as argument', async () => {
      await expect(
        execute(
          [
            'get',
            'pipeline-revision',
            '--project-id',
            knownIds.static.project,
            '--pipeline-id',
            knownIds.static.pipeline,
            knownIds.static.pipelineRevision
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });

    test('accepts id from stdin', async () => {
      await expect(
        executeAsIs(
          `echo ${knownIds.static.pipelineRevision} | ${withCmd(
            `get pipeline-revision --project-id ${knownIds.static.project} --pipeline-id ${knownIds.static.pipeline}`
          )}`,
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('pipeline', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'pipeline'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      await expect(
        execute(['get', 'pipeline', '--project-id', knownIds.static.project], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });

    test('accepts id as argument', async () => {
      await expect(
        execute(
          [
            'get',
            'pipeline',
            '--project-id',
            knownIds.static.project,
            knownIds.static.pipeline
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });

    test('accepts id from stdin', async () => {
      await expect(
        executeAsIs(
          `echo ${knownIds.static.pipeline} | ${withCmd(
            `get pipeline --project-id ${knownIds.static.project}`
          )}`,
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('runner', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'runner'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(
          ['get', 'runner', '--organization-id', knownIds.static.organization],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).rejects.toMatchSnapshot();
    });

    test('accepts id as argument', async () => {
      await expect(
        execute(
          [
            'get',
            'runner',
            '--organization-id',
            knownIds.static.organization,
            knownIds.static.runner
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });

    test('accepts id from stdin', async () => {
      await expect(
        executeAsIs(
          `echo ${knownIds.static.runner} | ${withCmd(
            `get runner --organization-id ${knownIds.static.organization}`
          )}`,
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('run', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'run'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(['get', 'run', '--project-id', knownIds.static.project], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });
});
