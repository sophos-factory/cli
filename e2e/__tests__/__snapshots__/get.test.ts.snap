// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refactrctl get credential accepts id as argument 1`] = `
"id                name                          type     data  created            
secret-test-cred  Super Secret Test Credential  generic  -     12/29/2021, 8:57 PM
"
`;

exports[`refactrctl get credential accepts id from stdin 1`] = `
"id                name                          type     data  created            
secret-test-cred  Super Secret Test Credential  generic  -     12/29/2021, 8:57 PM
"
`;

exports[`refactrctl get credential throws on missing arguments 1`] = `
"Usage: refactrctl get credential <credential-id> [options]

Positionals:
  credential-id  ID of the credential to fetch. Unlike IDs for other resources (which are mechanically generated), credential ID is user-provided reference ID  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project containing the credential  [string] [required]

Missing required argument: project-id
"
`;

exports[`refactrctl get credential throws on missing arguments 2`] = `
"Usage: refactrctl get credential <credential-id> [options]

Positionals:
  credential-id  ID of the credential to fetch. Unlike IDs for other resources (which are mechanically generated), credential ID is user-provided reference ID  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project containing the credential  [string] [required]

Credential ID must be provided
"
`;

exports[`refactrctl get job accepts id as argument 1`] = `
"_id                       name                      enabled  trigger_type  created            
61ccca11d42b9be6898365e0  How Do We Know It Works?  Yes      manual        12/29/2021, 8:50 PM
"
`;

exports[`refactrctl get job accepts id from stdin 1`] = `
"_id                       name                      enabled  trigger_type  created            
61ccca11d42b9be6898365e0  How Do We Know It Works?  Yes      manual        12/29/2021, 8:50 PM
"
`;

exports[`refactrctl get job throws on missing arguments 1`] = `
"Usage: refactrctl get job <job-id> [options]

Positionals:
  job-id  ID of the job to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project containing the job  [string] [required]

Missing required argument: project-id
"
`;

exports[`refactrctl get job throws on missing arguments 2`] = `
"Usage: refactrctl get job <job-id> [options]

Positionals:
  job-id  ID of the job to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project containing the job  [string] [required]

Job ID must be provided
"
`;

exports[`refactrctl get organization accepts id as argument 1`] = `
"_id                       name     contact_name    contact_email              created            
61cc93942529b6ec9681e4e2  CLI e2e  Victoria Drake  victoria.drake@sophos.com  12/29/2021, 4:57 PM
"
`;

exports[`refactrctl get organization accepts id from stdin 1`] = `
"_id                       name     contact_name    contact_email              created            
61cc93942529b6ec9681e4e2  CLI e2e  Victoria Drake  victoria.drake@sophos.com  12/29/2021, 4:57 PM
"
`;

exports[`refactrctl get organization throws on missing arguments 1`] = `
"Usage: refactrctl get organization <organization-id> [options]

Positionals:
  organization-id  ID of the organization to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]

Organization ID must be provided
"
`;

exports[`refactrctl get pipeline accepts id as argument 1`] = `
"_id                       organization_id           project_id                name                          step_count  revision_count  created            
61ccc379d42b9b0f518365db  61cc93942529b6ec9681e4e2  61ccbe03d42b9bdcc88365d8  The Answer is Always Testing  1           3               12/29/2021, 8:22 PM
"
`;

exports[`refactrctl get pipeline accepts id from stdin 1`] = `
"_id                       organization_id           project_id                name                          step_count  revision_count  created            
61ccc379d42b9b0f518365db  61cc93942529b6ec9681e4e2  61ccbe03d42b9bdcc88365d8  The Answer is Always Testing  1           3               12/29/2021, 8:22 PM
"
`;

exports[`refactrctl get pipeline throws on missing arguments 1`] = `
"Usage: refactrctl get pipeline <pipeline-id> [options]

Positionals:
  pipeline-id  ID of the pipeline to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project to fetch  [string] [required]

Missing required argument: project-id
"
`;

exports[`refactrctl get pipeline throws on missing arguments 2`] = `
"Usage: refactrctl get pipeline <pipeline-id> [options]

Positionals:
  pipeline-id  ID of the pipeline to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project to fetch  [string] [required]

Pipeline ID must be provided
"
`;

exports[`refactrctl get pipeline-revision accepts id as argument 1`] = `
"_id                       revision  created            
61ccc95a8997cfeb71742855  3         12/29/2021, 8:47 PM
"
`;

exports[`refactrctl get pipeline-revision accepts id from stdin 1`] = `
"_id                       revision  created            
61ccc95a8997cfeb71742855  3         12/29/2021, 8:47 PM
"
`;

exports[`refactrctl get pipeline-revision throws on missing arguments 1`] = `
"Usage: refactrctl get pipeline-revision <revision> [options]

Positionals:
  revision-id  ID of the pipeline revision to fetch  [string] [default: read from stdin]

Options:
      --version      Show version number  [boolean]
  -v, --verbose      Print detailed output  [boolean]
      --format       Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter       Filter output using JsonPath  [string]
      --address      Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token   Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help         Show help  [boolean]
      --project-id   ID of the project containing the pipeline revision  [string] [required]
      --pipeline-id  ID of the pipeline for the pipeline revision  [string] [required]

Missing required arguments: project-id, pipeline-id
"
`;

exports[`refactrctl get pipeline-revision throws on missing arguments 2`] = `
"Usage: refactrctl get pipeline-revision <revision> [options]

Positionals:
  revision-id  ID of the pipeline revision to fetch  [string] [default: read from stdin]

Options:
      --version      Show version number  [boolean]
  -v, --verbose      Print detailed output  [boolean]
      --format       Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter       Filter output using JsonPath  [string]
      --address      Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token   Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help         Show help  [boolean]
      --project-id   ID of the project containing the pipeline revision  [string] [required]
      --pipeline-id  ID of the pipeline for the pipeline revision  [string] [required]

Revision must be provided
"
`;

exports[`refactrctl get project accepts id as argument 1`] = `
"_id                       name          pipeline_count  organization_id           created            
61ccbe03d42b9bdcc88365d8  CRUD Testing  1               61cc93942529b6ec9681e4e2  12/29/2021, 7:58 PM
"
`;

exports[`refactrctl get project accepts id from stdin 1`] = `
"_id                       name          pipeline_count  organization_id           created            
61ccbe03d42b9bdcc88365d8  CRUD Testing  1               61cc93942529b6ec9681e4e2  12/29/2021, 7:58 PM
"
`;

exports[`refactrctl get project throws on missing arguments 1`] = `
"Usage: refactrctl get project <project-id> [options]

Positionals:
  project-id  ID of the project to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]

Project ID must be provided
"
`;

exports[`refactrctl get run throws on missing arguments 1`] = `
"Usage: refactrctl get run <run-id> [options]

Positionals:
  run-id  ID of the run to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project containing the run  [string] [required]
      --wait        Wait until run execution is finished if it is in progress  [boolean]

Missing required argument: project-id
"
`;

exports[`refactrctl get run throws on missing arguments 2`] = `
"Usage: refactrctl get run <run-id> [options]

Positionals:
  run-id  ID of the run to fetch  [string] [default: read from stdin]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]
      --project-id  ID of the project containing the run  [string] [required]
      --wait        Wait until run execution is finished if it is in progress  [boolean]

Run ID must be provided
"
`;

exports[`refactrctl get runner accepts id as argument 1`] = `
"_id                       organization_id           name            host_type  machine_type  version  started  heartbeat  status   idle_since  stop_pending  created            
61ccb65bd42b9b66fa8365d2  61cc93942529b6ec9681e4e2  CI Test Runner  -          -             -        -        -          Running  -           -             12/29/2021, 7:26 PM
"
`;

exports[`refactrctl get runner accepts id from stdin 1`] = `
"_id                       organization_id           name            host_type  machine_type  version  started  heartbeat  status   idle_since  stop_pending  created            
61ccb65bd42b9b66fa8365d2  61cc93942529b6ec9681e4e2  CI Test Runner  -          -             -        -        -          Running  -           -             12/29/2021, 7:26 PM
"
`;

exports[`refactrctl get runner throws on missing arguments 1`] = `
"Usage: refactrctl get runner <runner-id> [options]

Positionals:
  runner-id  ID of the runner to fetch  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
  -v, --verbose          Print detailed output  [boolean]
      --format           Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter           Filter output using JsonPath  [string]
      --address          Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner  [string] [required]

Missing required argument: organization-id
"
`;

exports[`refactrctl get runner throws on missing arguments 2`] = `
"Usage: refactrctl get runner <runner-id> [options]

Positionals:
  runner-id  ID of the runner to fetch  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
  -v, --verbose          Print detailed output  [boolean]
      --format           Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter           Filter output using JsonPath  [string]
      --address          Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner  [string] [required]

Runner ID must be provided
"
`;

exports[`refactrctl get throws on missing subcommand 1`] = `
"Usage: refactrctl get <command> [options]

Commands:
  refactrctl get credential [credential-id]       Get credential details
  refactrctl get job [job-id]                     Get job details
  refactrctl get project [project-id]             Get project details
  refactrctl get pipeline [pipeline-id]           Get pipeline details
  refactrctl get organization [organization-id]   Get organization details
  refactrctl get run [run-id]                     Get run details
  refactrctl get pipeline-revision [revision-id]  Get pipeline-revision details
  refactrctl get runner [runner-id]               Get runner details

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --filter      Filter output using JsonPath  [string]
      --address     Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS environment variable if set, otherwise https://api.refactr.it/v1]
      --auth-token  Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --help        Show help  [boolean]

Command must be specified
"
`;
